Add Kibana application privileges to the reserved role docs (#354)
elastic/stack-docs: docs/en/stack/security/authorization/built-in-roles.asciidoc

[[built-in-roles-data-frame-transforms-admin]] `data_frame_transforms_admin` ::
Grants `manage_data_frame_transforms` cluster privileges, which enable you to

> manage data frame transforms. This role also includes the `reserved_ml` {kib}

Reviewer: @kobelb will this level of Kibana privilege be documented anywhere?  It doesn't seem we go into that level of detail in  https://www.elastic.co/guide/en/kibana/master/kibana-privileges.html so maybe we should omit it here too?  e.g. Change it to something like "Users with this role have all {ml} feature privileges in Kibana".

Reviewer 2: That's a good point... we've generally hidden the concept of "Kibana reserved privileges" from end-users, so the phrasing which you've recommended might make this less confusing to users.

Reviewer: Thanks @kobelb !  @droberts195 I've drafted suggestions for the changes.

> manage data frame transforms. This role also includes the `reserved_ml` {kib}

Reviewer: ```suggestion
manage data frame transforms. This role also includes all
```

> application privilege which makes the {kib} {ml} application visible.

Reviewer: ```suggestion
{kibana-ref}/kibana-privileges.html[Kibana privileges] for the {ml-features}.
```

[[built-in-roles-data-frame-transforms-user]] `data_frame_transforms_user` ::
Grants `monitor_data_fram_transforms` cluster privileges, which enable you to

> use data frame transforms. This role also includes the `reserved_ml` {kib}

Reviewer: ```suggestion
use data frame transforms. This role also includes all
```

> application privilege which makes the {kib} {ml} application visible.

Reviewer: ```suggestion
{kibana-ref}/kibana-privileges.html[Kibana privileges] for the {ml-features}.
```

[[built-in-roles-ml-admin]] `machine_learning_admin`::
Grants `manage_ml` cluster privileges, read access to `.ml-anomalies*`,
`.ml-notifications*`, `.ml-state*`, `.ml-meta*` indices and write access to

> `.ml-annotations*` indices. This role also includes the `reserved_ml` {kib}

Reviewer: ```suggestion
`.ml-annotations*` indices. This role also includes all
```

> application privilege which makes the {kib} {ml} application visible.

Reviewer: ```suggestion
{kibana-ref}/kibana-privileges.html[Kibana privileges] for the {ml-features}.
```

[[built-in-roles-ml-user]] `machine_learning_user`::
Grants the minimum privileges required to view {ml} configuration,
status, and work with results. This role grants `monitor_ml` cluster privileges,
read access to the `.ml-notifications` and `.ml-anomalies*` indices
(which store {ml} results), and write access to `.ml-annotations*` indices.

> This role also includes the `reserved_ml` {kib} application privilege which

Reviewer: ```suggestion
This role also includes all {kibana-ref}/kibana-privileges.html[Kibana privileges] for the {ml-features}.
```

> makes the {kib} {ml} application visible.

Reviewer: ```suggestion

```

[[built-in-roles-monitoring-user]] `monitoring_user`::
Grants the minimum privileges required for any user of {monitoring} other than those
required to use {kib}. This role grants access to the monitoring indices and grants
privileges necessary for reading basic cluster information. This role also includes

> the `reserved_monitoring` {kib} application privilege which makes the {kib} monitoring

Reviewer: ```suggestion
all {kibana-ref}/kibana-privileges.html[Kibana privileges] for the {stack-monitor-features}.
```

> application visible. Monitoring users should also be assigned the `kibana_user` role.

Reviewer: ```suggestion
Monitoring users should also be assigned the `kibana_user` role.
```
